### PR TITLE
remove deprecations: `from` with scheduler, `combineLatest` with scheduler

### PR DESCRIPTION
**Pull request by benlesh**
Two of the more gnarly ones.

- `innerFrom` becomes `from`.
- `combineLatest` gets a LOT of crap removed from it. (there's more to go there, but that can be done in a follow up)

### CODE REVIEW

File: src/internal/observable/from.ts [hunk 1]
```diff
@@ -69,36 +75,117 @@ export function from<O extends ObservableInput<any>>(input: O, scheduler: Schedu
...
+  throw createInvalidObservableTypeError(input);
+}
+
+/**
+ * Creates an RxJS Observable from an object that implements `Symbol.observable`.
+ * @param obj An object that properly implements `Symbol.observable`.
  */
-export function from<T>(input: ObservableInput<T>, scheduler?: SchedulerLike): Observable<T> {
-  return scheduler ? scheduled(input, scheduler) : innerFrom(input);
+export function fromInteropObservable<T>(obj: any) {
```
Comment (jakovljevic-mladen) [inline src/internal/observable/from.ts:119]: Do these functions need to be exported? They don't seem to be used anywhere else.

Comment (benlesh) [reply src/internal/observable/from.ts]: Good call. Done.

**Comment by benlesh** [comment]
Reviewers: You'll want to review this commit by commit to make it easier on yourselves.

File: src/internal/observable/from.ts [hunk 2]
```diff
@@ -65,36 +75,117 @@ export function from<O extends ObservableInput<any>>(input: O): Observable<Obser
...
- * ```
- *
  * @see {@link fromEvent}
  * @see {@link fromEventPattern}
+ * @see {@link scheduled}
  *
- * @param {ObservableInput<T>} A subscription object, a Promise, an Observable-like,
- * an Array, an iterable, or an array-like object to be converted.
- * @param {SchedulerLike} An optional {@link SchedulerLike} on which to schedule the emission of values.
- * @return {Observable<T>}
```
Comment (jakovljevic-mladen) [inline src/internal/observable/from.ts:100]: I'd keep this `@return`. Something like this:

```
@return An Observable converted from the `input`.
```

File: src/internal/observable/from.ts [hunk 3]
```diff
@@ -65,36 +75,117 @@ export function from<O extends ObservableInput<any>>(input: O): Observable<Obser
...
+ * @param input A subscription object, a Promise, an Observable-like,
+ * an Array, an iterable, async iterable, or an array-like object to be converted.
```
Comment (jakovljevic-mladen) [inline src/internal/observable/from.ts:83]: `async iterable` may need an article: `an async iterable`.

File: src/internal/observable/from.ts [hunk 4]
```diff
@@ -1,8 +1,18 @@
...
+import { isIterable } from '../util/isIterable';
+import { isReadableStreamLike, readableStreamLikeToAsyncGenerator } from '../util/isReadableStreamLike';
+import { Subscriber } from '../Subscriber';
+import { isFunction } from '../util/isFunction';
+import { reportUnhandledError } from '../util/reportUnhandledError';
+import { observable as Symbol_observable } from '../symbol/observable';
 
-export function from<O extends ObservableInput<any>>(input: O): Observable<ObservedValueOf<O>>;
 /**
  * Creates an Observable from an Array, an array-like object, a Promise, an iterable object, or an Observable-like object.
```
Comment (jakovljevic-mladen) [inline src/internal/observable/from.ts:17]: I think we should cover all [ObservableInput](https://rxjs.dev/api/index/type-alias/ObservableInput) here. Also, `an Array` and `an array-like` are the same I guess, so I think that these can be merged to one.

If this is not relevant for this PR, we can fix it in another one. If you decide to include such a fix here, please update `@param input` on the bottom of the JSDoc as well.

**Comment by benlesh** [comment]
@jakovljevic-mladen maybe we can have a follow-up to revamp the `from` docs? Right now I'd just like to get the removal landed, as this one effects a lot of stuff.

**Comment by jakovljevic-mladen** [comment]
@benlesh, I agree. I had a plan to write much more `ObservableInput` docs, we have some open issues regarding that. I could cover both `from` and `ObservableInput` in a single PR.

**Review by jakovljevic-mladen — APPROVED** [review]
LGTM, only minor docs issues that _could_ be considered to be fixed right away (an article and `@return`).

File: src/internal/observable/of.ts [hunk 5]
```diff
@@ -79,5 +80,5 @@ export function of<A extends readonly unknown[]>(...values: A): Observable<Value
  */
 export function of<T>(...args: Array<T | SchedulerLike>): Observable<T> {
   const scheduler = popScheduler(args);
-  return from(args as T[], scheduler);
+  return scheduler ? scheduled(args as T[], scheduler) : from(args as T[]);
```
Comment (demensky) [inline src/internal/observable/of.ts:83]: Maybe `fromArrayLike` would be better here?

```suggestion
  return scheduler ? scheduled(args as T[], scheduler) : fromArrayLike(args as T[]);
```

File: src/internal/observable/of.ts [hunk 6]
```diff
@@ -79,5 +80,5 @@ export function of<A extends readonly unknown[]>(...values: A): Observable<Value
  */
 export function of<T>(...args: Array<T | SchedulerLike>): Observable<T> {
   const scheduler = popScheduler(args);
-  return from(args as T[], scheduler);
+  return scheduler ? scheduled(args as T[], scheduler) : fromArrayLike(args as T[]);
```
Comment (jeremymwells) [inline src/internal/observable/of.ts:83]: `fromArrayLike` is not exported from `from`. Should this just be `from`?

Comment (demensky) [reply src/internal/observable/of.ts:83]: There can be nothing here but an array. `from` [contains a lot of code](https://github.com/ReactiveX/rxjs/blob/314b204acc9a8923830789c948727febb5b521e1/src/internal/observable/from.ts#L92-L109) that converts various types into `Observable` which will never be executed in the context of `of`.

**Review by jeremymwells — APPROVED** [review]

**Pull request merged**